server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceConfig.java

```diff
@@ -269,8 +275,36 @@ public ValuesSourceConfig(
         this.unmapped = unmapped;
         this.script = script;
         this.scriptValueType = scriptValueType;
+        this.missing = missing;
+        this.timeZone = timeZone;
+        this.format = format == null ? DocValueFormat.RAW : format;
         this.nowSupplier = nowSupplier;
 
+        if (!valid()) {
+            // TODO: resolve no longer generates invalid configs.  Once VSConfig is immutable, we can drop this check
+            throw new IllegalStateException(
+                "value source config is invalid; must have either a field context or a script or marked as unwrapped");
+        }
+
+        final ValuesSource vs;
```

Comment:
This'd probably be a little cleaner as a `private final` method with early returns.